docs/en/codebase_en.md

```diff
@@ -0,0 +1,89 @@
+# Code Documentation
+
+This document is intended to facilitate reading the codebase of the MyUoM application and for people who would like to help by writing code or (debugging).
+
+Contents:
+
+- Project Libraries
+- Application Architecture
+- Folder Structure
+- Quick Start Guide( start contributing )
+- Coding Conventions
+- Code Analysis
+  - Contexts
+  - Hooks
+
+### Libraries project
+
+For the frontend we used reactjs along with [Chakra UI](https://chakra-ui.com/) as a library for ready-made components and styling (it also provides various tools and some icons).
+For the animations we also used the framer-motion library(dependency of Chakra UI which we needed additionally). For routing the library [react router dom v6](https://reactrouter.com/en/v6.3.0/getting-started/overview) was utilized.
+The application started using [create-react-app](https://create-react-app.dev/) for convenience(but will soon be changed to Vite.js, and typescript).
+For some additional icons the libraries [react-icons](https://react-icons.github.io/react-icons/) and [react-social-icons](https://www.npmjs.com/package/react-social-icons) were also used.
+
+### Run the code
+
+For now because it's create-react-app, the commands needed are:
+`git clone https://gitlab.com/opensourceuom/myUoM.git`
+
+`npm install` (puts the necessary libraries if node.js and npm are present otherwise they will need to be downloaded separately)
+
+`npm start` runs the application on localhost:3000, when the change to Vite.js is made it will be npm run dev
```

Comment:
```suggestion
`npm start` runs the application on [localhost:3000](http://localhost:3000), when the change to Vite.js is made it will be `npm run dev`
```
it looks good